sphinxcontrib/autodoc_pydantic/inspection.py

```diff
@@ -462,8 +462,11 @@ def is_pydantic_model(obj: Any) -> bool:
 
         """
 
-        if isinstance(obj, type):
+        try:
             return issubclass(obj, BaseModel)
+        except TypeError:
+            pass
+
         return False
```

Comment:
The unnested fallback return value is a mypy user habit
I believe your revision would pass pyright which does deeper static analysis but mypy tends to be less clever and wants a return at the end of each function

Reply:
Thanks for the explanation. I haven't been using mypy a lot and `autodoc_pydantic` is currently not type checking safe. As for now, the type annotations mainly help to reason about the code but it would be very benefical to add mypy/pyright in the future.